c7n/resources/networkmanager.py

```diff
@@ -186,3 +186,54 @@ def process(self, resources):
                 client.delete_global_network(GlobalNetworkId=r['GlobalNetworkId'])
             except client.exceptions.ResourceNotFoundException:
                 continue
+
+
+@Site.action_registry.register('delete')
+class DeleteSite(BaseAction):
+    """Action to delete a networkmanager site
+    """
+    schema = type_schema('delete')
+    permissions = ('networkmanager:DeleteSite',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('networkmanager')
+
+        for r in resources:
+            try:
+                client.delete_site(GlobalNetworkId=r['GlobalNetworkId'], SiteId=r['SiteId'])
+            except client.exceptions.ResourceNotFoundException:  # pragma: no cover
+                continue
+
+
+@Device.action_registry.register('delete')
+class DeleteDevice(BaseAction):
+    """Action to delete a networkmanager device
+    """
+    schema = type_schema('delete')
+    permissions = ('networkmanager:DeleteDevice',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('networkmanager')
+
+        for r in resources:
+            try:
+                client.delete_device(GlobalNetworkId=r['GlobalNetworkId'], DeviceId=r['DeviceId'])
+            except client.exceptions.ResourceNotFoundException:  # pragma: no cover
+                continue
+
+
+@Link.action_registry.register('delete')
```

Comment:
The link needs to be disassociated from any devices and customer gateway before it can be deleted